third_party/move/move-core/types/src/value.rs

```diff
@@ -722,13 +826,20 @@ impl fmt::Display for MoveStruct {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         match self {
             MoveStruct::Runtime(v) => fmt_list(f, "struct[", v, "]"),
+            MoveStruct::RuntimeVariant(tag, v) => fmt_list(f, &format!("variant#{}[", tag), v, "]"),
             MoveStruct::WithFields(fields) => {
                 fmt_list(f, "{", fields.iter().map(DisplayFieldBinding), "}")
             },
             MoveStruct::WithTypes { type_, fields } => {
                 fmt::Display::fmt(type_, f)?;
                 fmt_list(f, " {", fields.iter().map(DisplayFieldBinding), "}")
             },
+            MoveStruct::WithVariant(name, _tag, fields) => fmt_list(
+                f,
+                &format!("{}{{", name),
```

Comment:
Should we format the name like you did in the resource viewer, with `::name { ... } `? I think a space before `{{` is missing?